pkgs/development/libraries/protobuf/generic-v3-cmake.nix

```
@@ -80,6 +80,8 @@ let
       "-Dprotobuf_ABSL_PROVIDER=package"
     ] ++ lib.optionals (!stdenv.targetPlatform.isStatic) [
       "-Dprotobuf_BUILD_SHARED_LIBS=ON"
+    ] ++ lib.optionals (stdenv.cc.cc.defaultCxxStandard < 14) [
+      "-DCMAKE_CXX_STANDARD=14"
```

```suggestion
    ] ++ lib.optionals (stdenv.cc.cc.defaultCxxStandard < 14) [
      "-DCMAKE_CXX_STANDARD=14"
```

What??

How can `stdenv.cc.cc.defaultCxxStandard` be less than 14 but `-DCMAKE_CXX_STANDARD`  be 14.

---

The idea here is to pass `-DCMAKE_CXX_STANDARD` only if the compiler in use defaults to an older language version. C++14 is the minimum required to build protobuf.

---

Then you should  wrote
```suggestion
    ] ++ lib.optionals (stdenv.cc.cc.defaultCxxStandard >= 14) [
      "-DCMAKE_CXX_STANDARD=14"
```

---

Oh, I understood. The default C++ standard is not the same as the supported C++ standard.

However it looks too superfluous. 
"If the default C++ standard of this compiler is not 14, then force-set the C++ standard to 14".
Why not set the C++ standard to 14 and ignore the compiler's default?

---

If the default is newer than 14 some of the dependencies are not API compatible (cough, abseil, cough), so that approach doesn't work.

---

We should write a comment to explain that so that the next guy who looks as this doesn't go through the same process.
something like
```
# protobuf needs at least c++14 to be compiled
# if the default is higher don't override it as abseil for example won't be api compatible
```